src/scrypt_fmt.c

```diff
@@ -243,7 +243,8 @@ static int valid(char *ciphertext, struct fmt_main *self)
 static void *get_binary(char *ciphertext)
 {
 	static char out[BINARY_SIZE];
-	strncpy(out, ciphertext, sizeof(out)); /* NUL padding is required */
+	strncpy(out, ciphertext, sizeof(out)-1);
+	out[sizeof(out)-1] = 0;
```

Comment:
Please at least re-introduce the comment.

Reply:
I will be reverting this soon.